[0-th enum value JSON serialization] [part 1/2] Unit tests

antongrbin opened the pull request:
# Motivation

See https://github.com/streem/pbandk/issues/235#issue-1416975668.

This PR only adds the unit tests that document the current behavior. We will implement the fix as discussed in #235 in the follow-up PR, after this one is merged.

# Changes

* Added a unit test to document the behavior that will be fixed in a follow-up PR.
* After changing the `test_proto2.proto` I executed `./gradlew generateProtos` to generate the `TestProto.java` and `test_proto2.kt`

# Tested

Executed unit tests locally and they pass.

garyp commented on runtime/src/commonTest/kotlin/pbandk/json/JsonTest.kt:
```diff
@@ -18,6 +19,40 @@ class JsonTest {
+        // This behavior is unexpected and will be fixed in a follow-up.
+        // See https://github.com/streem/pbandk/issues/235 for more details.
+        val message = MessageWithEnum()
+        assertEquals("{\"value\":null}", message.encodeToJsonString(jsonConfig))
```
FWIW, this is compliant with the spec since `null` and not-present values in the JSON are supposed to be treated equivalently in the protobuf JSON encoding. But I agree that it'd be better if the `null` value is not output unless `JsonConfig.outputDefaultValues=true`.

antongrbin replied:
Agreed that the `null` doesn't hurt. I decided to remove the comments.

antongrbin replied:
Out of curiosity I tested official c++, python and java to see if they will output a `null` value in this scenario (by default outputting default values is set to false):
https://github.com/protocolbuffers/protobuf/pull/10812

The conclusion is that they don't emit an explicit `null`.

garyp reviewed (approved):
The tests look good 👍🏼 Just a couple style-related comments to address before merging.

antongrbin commented on runtime/src/commonTest/kotlin/pbandk/json/JsonTest.kt:
```diff
@@ -18,6 +19,40 @@ class JsonTest {
+
+        // This works as expected.
+        val messageWith1 = MessageWithEnum(value = MessageWithEnum.EnumType.BAR)
+        assertEquals("{\"value\":\"BAR\"}", messageWith1.encodeToJsonString(jsonConfig))
```
Agreed 100%. Done!

antongrbin reviewed (commented):
Thank you for the review!

antongrbin commented on runtime/src/commonTest/kotlin/pbandk/json/JsonTest.kt:
```diff
@@ -18,6 +19,40 @@ class JsonTest {
     }
 
+    @Test
+    fun testMessageWithEnumProto2() {
```
Moved!

antongrbin commented:
@garyp I addressed the comments, thank you for your time.

Note that the merging is blocked on workflow waiting approval (first time contributor here).
